test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Config
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class EdgeConfigBuilder
+    {
+        readonly string deviceId;
+        readonly Dictionary<string, IModuleConfigBuilder> moduleBuilders;
+        Option<(string address, string username, string password)> registry;
+
+        public EdgeConfigBuilder(string deviceId)
+        {
+            this.deviceId = deviceId;
+            this.moduleBuilders = new Dictionary<string, IModuleConfigBuilder>();
+            this.registry = Option.None<(string, string, string)>();
+        }
+
+        public void AddRegistryCredentials(string address, string username, string password)
+        {
+            this.registry = Option.Some((address, username, password));
+        }
+
+        public IModuleConfigBuilder AddEdgeAgent(string image = null)
+        {
+            Option<string> imageOption = Option.Maybe(image);
+            var builder = new AgentModuleConfigBuilder(imageOption);
+            this.moduleBuilders.Add(builder.Name, builder);
+            return builder;
+        }
+
+        public IModuleConfigBuilder AddEdgeHub(string image = null)
```

Comment:
Since EdgeAgent and EdgeHub module are system modules, should it be better to store in another explicit variable and check if they exists during build method call?

Reply:
Edge hub is a system module, but it isn't required. When the test calls `builder.Build()` I make sure edge agent is present, and if not I add it. So this is already handled.

Reply:
Good to know. Why allow edgeHub to be missing? from azure portal UI, it looks it is a must.

Reply:
The portal is optimized for the common cases, which use edge hub. But it isn't strictly necessary, even in the portal. If a customer just wants to use edge to manage their devices from that cloud and they aren't concerned about sending telemetry to IoT Hub, then they don't need edge hub in their deployment.